core/src/main/scala/com/metamx/tranquility/config/ConfigHelper.scala

```diff
@@ -0,0 +1,114 @@
+package com.metamx.tranquility.config
+
+import java.io.InputStream
+import java.util.Properties
+
+import com.metamx.common.scala.Predef._
+import com.metamx.common.scala.Yaml
+import com.metamx.common.scala.untyped._
+import com.metamx.tranquility.druid.DruidBeams
+import com.metamx.tranquility.druid.DruidEnvironment
+import com.metamx.tranquility.druid.DruidGuicer
+import com.metamx.tranquility.druid.DruidLocation
+import com.metamx.tranquility.finagle.FinagleRegistry
+import com.metamx.tranquility.tranquilizer.Tranquilizer
+import io.druid.segment.realtime.FireDepartment
+import org.apache.curator.framework.CuratorFramework
+import org.skife.config.ConfigurationObjectFactory
+
+/**
+ * Helper methods to perform shared tasks based on common configuration file semantics.
+ */
+object ConfigHelper
+{
+  def readConfigYaml[T <: TranquilityConfig](
+    in: InputStream,
+    clazz: Class[T]
+    ): (T, Map[String, DataSourceConfig[T]], Dict) =
+  {
+    val yaml = in.withFinally(_.close()) { in =>
+      dict(Yaml.load(in))
+    }
+    val globalProperties: Dict = getAsDict(yaml, "properties")
+    val globalConfig = mapConfig(globalProperties, clazz)
+
+    for (prop <- globalProperties.filterKeys { _ startsWith("druid.extensions.") }) {
+      sys.props += (prop._1 -> prop._2.toString)
```

Comment:
Hmm, any editing of system properties is kind of gnarly. It's probably ok since it's meant to be called by a main, but would suggest two things:

- Comment on `readConfigYaml` indicating that it may modify system properties
- Only set the system property if it's not already set. This is because we usually want to respect a property from the commandline over one supplied in a config file.